src/NServiceBus.Core/Utils/ExceptionHeaderHelper.cs

```diff
@@ -1,55 +1,47 @@
 ﻿namespace NServiceBus
 {
     using System;
-    using System.Collections;
     using System.Collections.Generic;
     using System.Configuration;
 
     static class ExceptionHeaderHelper
     {
-        public static void SetExceptionHeaders(Dictionary<string, string> headers, Exception e)
+        public static void SetExceptionHeaders(Dictionary<string, string> headers, ExceptionInfo exceptionInfo)
         {
-            SetExceptionHeaders(headers, e, useLegacyStackTrace);
+            SetExceptionHeaders(headers, exceptionInfo, useLegacyStackTrace);
         }
 
-        public static void SetExceptionHeaders(Dictionary<string, string> headers, Exception e, bool legacyStacktrace)
+        public static void SetExceptionHeaders(Dictionary<string, string> headers, ExceptionInfo exceptionInfo, bool legacyStacktrace)
         {
-            headers["NServiceBus.ExceptionInfo.ExceptionType"] = e.GetType().FullName;
+            headers["NServiceBus.ExceptionInfo.ExceptionType"] = exceptionInfo.TypeFullName;
 
-            if (e.InnerException != null)
+            if (exceptionInfo.InnerExceptionTypeFullName != null)
             {
-                headers["NServiceBus.ExceptionInfo.InnerExceptionType"] = e.InnerException.GetType().FullName;
+                headers["NServiceBus.ExceptionInfo.InnerExceptionType"] = exceptionInfo.InnerExceptionTypeFullName;
             }
 
-            headers["NServiceBus.ExceptionInfo.HelpLink"] = e.HelpLink;
-            headers["NServiceBus.ExceptionInfo.Message"] = e.GetMessage().Truncate(16384);
-            headers["NServiceBus.ExceptionInfo.Source"] = e.Source;
+            headers["NServiceBus.ExceptionInfo.HelpLink"] = exceptionInfo.HelpLink;
+            headers["NServiceBus.ExceptionInfo.Message"] = exceptionInfo.Message.Truncate(16384);
+            headers["NServiceBus.ExceptionInfo.Source"] = exceptionInfo.Source;
 
             if (legacyStacktrace)
```

Comment:
@SimonCropp our proposal would be to deprecate it entirely. The app.config key is not documented.